src/metabase/models/permissions.clj

```diff
@@ -929,13 +721,15 @@
                                                            [:like :object (h2x/literal "%/db/%")]])
                                 (m/map-vals (fn [paths]
                                               ;; remove v1 paths, implicitly keep v2 paths
-                                              (remove (fn [path] (mc/validate [:re (u.regex/rx "^/" v1-data-permissions-rx "$")]
+                                              (remove (fn [path] (mc/validate [:re (u.regex/rx
+                                                                                    "^/"
+                                                                                    perms.u/v1-data-permissions-rx
+                                                                                    "$")]
```

Comment:
you didn't change this, but I noticed it's rebuilding 

```
(u.regex/rx "^/" v1-data-permissions-rx "$")
```

on each iteration of remove

Reply:
Changing to use `perms.u/DataPath`